code/modules/events/fugitive_spawning.dm

```diff
@@ -9,113 +9,118 @@
 	var/datum/mind/player_mind = new /datum/mind(selected.key)
 	player_mind.active = TRUE
 	var/mob/living/carbon/human/S = new(landing_turf)
 	player_mind.transfer_to(S)
-	player_mind.assigned_role = "Fugitive"
-	player_mind.special_role = "Fugitive"
-	player_mind.add_antag_datum(/datum/antagonist/fugitive)
-	var/datum/antagonist/fugitive/fugitiveantag = player_mind.has_antag_datum(/datum/antagonist/fugitive)
-	INVOKE_ASYNC(fugitiveantag, /datum/antagonist/fugitive.proc/greet, backstory) //some fugitives have a sleep on their greet, so we don't want to stop the entire antag granting proc with fluff
+	player_mind.assigned_role = ROLE_FUGITIVE
+	player_mind.special_role = ROLE_FUGITIVE
+	var/datum/antagonist/fugitive/A = new()
+	A.backstory = backstory
+	player_mind.add_antag_datum(A)
+	var/outfit = (leader && backstory.has_leader) ? backstory.leader_outfit : backstory.outfit
+	if(islist(outfit))
+		outfit = outfit[((index - 1) % length(outfit)) + 1]
+	S.equipOutfit(outfit)
 
-	switch(backstory)
-		if("prisoner")
-			S.equipOutfit(/datum/outfit/prisoner)
-		if("cultist")
-			S.equipOutfit(/datum/outfit/yalp_cultist)
-		if("waldo")
-			S.equipOutfit(/datum/outfit/waldo)
-		if("synth")
-			S.equipOutfit(/datum/outfit/synthetic)
 	message_admins("[ADMIN_LOOKUPFLW(S)] has been made into a Fugitive by an event.")
 	log_game("[key_name(S)] was spawned as a Fugitive by an event.")
-	spawned_mobs += S
 	return S
 
- //special spawn for one member. it can be used for a special mob or simply to give one normal member special items.
-/datum/round_event/ghost_role/fugitives/proc/gear_fugitive_leader(var/mob/dead/leader, var/turf/landing_turf, backstory)
-	var/datum/mind/player_mind = new /datum/mind(leader.key)
-	player_mind.active = TRUE
-	//if you want to add a fugitive with a special leader in the future, make this switch with the backstory
-	var/mob/living/carbon/human/S = gear_fugitive(leader, landing_turf, backstory)
-	var/obj/item/choice_beacon/augments/A = new(S)
-	S.put_in_hands(A)
-	new /obj/item/autosurgeon(landing_turf)
-
-//security team gets called in after 10 minutes of prep to find the refugees
-/datum/round_event/ghost_role/fugitives/proc/spawn_hunters()
-	var/backstory = pick("space cop", "russian", "bounty hunter")
-	var/datum/map_template/shuttle/ship
-	if(backstory == "space cop")
-		ship = new /datum/map_template/shuttle/hunter/space_cop
-	else if (backstory == "russian")
-		ship = new /datum/map_template/shuttle/hunter/russian
-	else
-		ship = new /datum/map_template/shuttle/hunter/bounty
+// Security team gets called in after 10 minutes of prep to find the fugitives
+/proc/spawn_hunters()
+	var/datum/fugitive_type/hunter/backstory = GLOB.hunter_types[pick(GLOB.hunter_types)]
+	var/list/candidates = pollGhostCandidates("The Fugitive Hunters are looking for a [backstory.name]. Would you like to be considered for this role?", ROLE_FUGITIVE_HUNTER)
+	var/datum/map_template/shuttle/ship = new backstory.ship_type
```

Comment:
Use the cached maps rather than making a new one, I think SSmapping stores the map files. If not ignore